src/server/game/Server/Protocol/Opcodes.cpp

```diff
@@ -323,7 +323,7 @@ void OpcodeTable::Initialize()
     DEFINE_HANDLER(CMSG_CLEAR_RAID_MARKER,                                  STATUS_LOGGEDIN,  PROCESS_THREADUNSAFE, &WorldSession::HandleClearRaidMarker);
     DEFINE_HANDLER(CMSG_CLEAR_TRADE_ITEM,                                   STATUS_LOGGEDIN,  PROCESS_THREADUNSAFE, &WorldSession::HandleClearTradeItemOpcode);
     DEFINE_HANDLER(CMSG_CLIENT_PORT_GRAVEYARD,                              STATUS_LOGGEDIN,  PROCESS_THREADUNSAFE, &WorldSession::HandlePortGraveyard);
-    DEFINE_HANDLER(CMSG_CLOSE_INTERACTION,                                  STATUS_UNHANDLED, PROCESS_THREADSAFE,   &WorldSession::HandleCloseInteraction);
+    DEFINE_HANDLER(CMSG_CLOSE_INTERACTION,                                  STATUS_LOGGEDIN, PROCESS_THREADSAFE,   &WorldSession::HandleCloseInteraction);
```

Comment:
change `PROCESS_THREADSAFE` to `PROCESS_THREADUNSAFE` too and it can fix your not responding npcs, but anyway in that packet we should only reset the saved guid if it matches the sent guid

Reply:
`PROCESS_THREADUNSAFE` don't fix this. Remember: we need a trainer with gossip for try it (not with creature_default_trainer)

Reply:
Can you post an entry of such a creature if one exists? to test with.

Reply:
https://gist.github.com/DoctorKraft/a9b6562c838e2f23a0ae34c59232ca8f

Reply:
trainerlist, gossiphello, gossipselect are processes threadunsafe, this opcode must also processed threadunsafe to ensure the right order of received packets like the client sends them, closeinteraction before gossiphello and others